engine/src/main/java/org/terasology/world/block/regions/BlockRegionSystem.java

```diff
@@ -38,10 +43,25 @@
     @In
     private BlockManager blockManager;
 
-    @ReceiveEvent
+    // trivial priority so that all other logic can happen to the region before erasing the blocks in the region
+    @ReceiveEvent(priority = EventPriority.PRIORITY_TRIVIAL)
     public void onDestroyed(DoDestroyEvent event, EntityRef entity, BlockRegionComponent blockRegion) {
         for (Vector3i blockPosition : blockRegion.region) {
             worldProvider.setBlock(blockPosition, blockManager.getBlock(BlockManager.AIR_ID));
         }
     }
+
+    @ReceiveEvent
+    public void onActivatedActAsBlockAddHealth(OnActivatedComponent event, EntityRef entity, BlockRegionComponent blockRegion, ActAsBlockComponent actAsBlock) {
```

Comment:
This is to help a module developer maintain consistency for block health values.  Currently you would have to ensure that both the block and the region's entity are synchronized to get a similar feel.  This would get even more tricky if a block was overridden by another module.

I agree that block health really needs to get pulled out of engine to avoid polluting the engine code with this kind of gameplay specific code.